src/Hint/Fixities.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE ViewPatterns, ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
+{-
+
+Raise a warning if you have redundant brackets in nested infix expressions.
+
+<TEST>
+yes = 1 + (2 * 3) -- @Ignore 1 + 2 * 3
+no = (1 + 2) * 3
+no = 1 + 2 * 3
+</TEST>
+-}
+
+module Hint.Fixities(fixitiesHint) where
+
+import Hint.Type(DeclHint,Idea(..),rawIdea,Severity(..),toSS)
+import Config.Type
+import Data.Data(Data)
+import Data.List.Extra
+import Data.Map(Map, singleton, fromList)
+import Data.Generics.Uniplate.DataOnly
+import Fixity
+import Refact.Types
+
+import GHC.Hs
+import Outputable
+import SrcLoc
+import GHC.Util
+import Language.Haskell.GhclibParserEx.GHC.Utils.Outputable
+
+fixitiesHint :: [Setting] -> DeclHint
+fixitiesHint settings _ _ x =
+  concatMap (infixBracket fixities) (childrenBi x :: [LHsExpr GhcPs]) ++
+  concatMap (infixBracket fixities) (childrenBi x :: [LHsType GhcPs]) ++
+  concatMap (infixBracket fixities) (childrenBi x :: [LPat GhcPs])
+   where
+     fixities = foldMap getFixity settings `mappend` fromList (map toFixity defaultFixities)
+     getFixity (Infix x) = uncurry singleton (toFixity x)
+     getFixity _ = mempty
+
+infixBracket :: forall a . (Data a, HasSrcSpan a, Outputable a, Brackets a) => Map String Fixity -> a -> [Idea]
```

Comment:
I think this could be just `HsExpr` rather than `a`, which simplifies everything.

Reply:
Done